orbit/pkg/update/runner.go

```diff
@@ -144,7 +146,21 @@ func (r *Runner) HasLocalHash(target string) bool {
 func (r *Runner) Execute() error {
 	log.Debug().Msg("start updater")
 
-	ticker := time.NewTicker(r.opt.CheckInterval)
+	// Randomize the initial interval so that all agents don't synchronize their updates
+	initialInterval := r.opt.CheckInterval
+	// Assuming CheckInterval > 0 because we validated it in NewRunner
+	nBig, err := rand.Int(rand.Reader, big.NewInt(int64(r.opt.CheckInterval)))
```

Comment:
Was rand already seeded in the code?
If not need to 'seed' it.

Reply:
I'm using the now-recommended `crypto/rand`, which does not have/need `seed`. Also, the `seed` in `math/rand` has been deprecated and no longer needed as of Go 1.20: https://pkg.go.dev/math/rand#:~:text=Deprecated%3A%20As%20of%20Go%201.20,obtain%20a%20local%20random%20generator.

I tested this, and I'm always getting different random values.

Reply:
Got it. wasn't updated on this :)